geomstats/geometry/matrices.py

```diff
@@ -67,40 +93,49 @@ def transpose(mat):
         -------
         transpose : array-like, shape=[n_samples, dim, dim]
         """
-        is_vec = (gs.ndim(gs.array(mat)) == 3)
-        axes = (0, 2, 1) if is_vec else (1, 0)
+        is_vectorized = (gs.ndim(gs.array(mat)) == 3)
```

Comment:
this is the second time I see this in the PR, maybe consider refactoring into a method?

Reply:
what do you mean? an `is_vectorized` method?

Reply:
... this might be a great idea, especially for unification of vectorization accross classes!  :smiley:

Reply:
Yes! To be put in the `vectorization.py` in this case.

Reply:
Thanks for the idea stefan I really think it's good! 
I'll keep it like as it is for now but let's keep on thinking about it in #338

Reply:
This `class.base_shape` attribute would be especially usefull for handling higher dimensional vectorization, for instance when computing 4-D arrays of geodesics, e.g. returning `[n_samples, time, dim1, dim2]` shape

Reply:
Yes! Currently, it is provided by an object's attribute called `default_point_type` or `point_type`. I agree that having it as a class attribute would probably be better if default computations are always in a certain format for the whole class. Happy to rename it `base_shape` or `default_shape` (probably `base_ndim` though?) if it's more intuitive, but let's rename it everywhere in this case to avoid namings' complexity explosion 💥